### PR TITLE
Attach the xhr violation error to the form field

### PR DESCRIPTION
**Pull request by ggabrovski**
## Attach the xhr violation error to the form field

In relation to @dmitryuk fix for #7945 #7946 it will be nice for the user to see the violation error message attached to the field. I suggest to add small logic to display this information on the right place.

## Changelog

```markdown
### Fixed
- Violations error handling
```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Can you provide a screenshot of what you get ?

**Comment by ggabrovski** [comment]
In the moment if you have ModelAutocompleteType with add button. When you try to add new record validations error are not show. @dmitryuk fix #7946 will show the validation error on top. 
Like this:

befor

I suggest to attach the violation error to the related form field:

after

**Review by VincentLanglet — DISMISSED** [review]

**Comment by dmitryuk** [comment]
Guys please merge PR and add a tag

**Comment by VincentLanglet** [comment]
I'd like the review of @jordisala1991 first

File: src/Resources/views/CRUD/Association/edit_many_script.html.twig [hunk 1]
```diff
@@ -361,18 +361,36 @@ This code manages the many-to-[one|many] association field popup
...
+                    $(form).find('.xhr-violation-error-block').remove();
+
+                    if (jsonContent.violations) {
+                        for (violation of jsonContent.violations) {
+                            // Try to find and attach the error to the form field, if not show message on top.
+                            if (violation.hasOwnProperty('propertyPath')) {
+                                const field = $(form).find(`[name='${violation.propertyPath}']`);
+                                if (field.length) {
+                                    field.closest('.form-group').addClass('has-error xhr-violation-error');
+                                    field.parent().append(`<div class="help-block sonata-ba-field-error-messages xhr-violation-error-block">
```
Comment (jordisala1991) [inline src/Resources/views/CRUD/Association/edit_many_script.html.twig]: This won't work if one field triggers 2 validation errors. IMO this code should be written in plain JS, and you can use some (or all) the logic written here: https://github.com/sonata-project/SonataPageBundle/blob/4.x/assets/js/composer.js#L255-L300

It should be quite similar.

**Comment by ggabrovski** [comment]
Hi @jordisala1991,
thanks for your help and guidance. Can I ask you to check it, please.

**Comment by dmitryuk** [comment]
@jordisala1991 please check the pr

**Review by jordisala1991 — APPROVED** [review]
IMO it looks good now, but we also need to move symfony/serializer from require-dev to require. It is a hard dependency

**Comment by jordisala1991** [comment]
Thank you @ggabrovski

**Pull request merged**